roles/generic-system-hardening/README.md

```diff
@@ -16,9 +16,55 @@
 
 ## Input data
 
-- the TCP port(s) the SSH server listens on (default: only port 22);
-- a list of allowed SSH logins (default: only the user used as Ansible login);
-- a list of logins allowed to use `sudo` (default: only the user used as Ansible
-  login);
-- an encrypted login password for each user (default: no password is set if not
-  specified).
+### Required
+
+##### `deflect_ssh_users`
+
+List of users allowed to SSH into the target system. Each item of the list
+represents a user and supports the following keys:
+
+- `login` (string) (mandatory): the user's login on the system;
+- `sudo` (boolean) (default: `False`): whether or not the user has `sudo`
+  access;
+- `password` (string): encrypted UNIX password for the user, following the
+  `/etc/shadow` format (see the `mkpasswd` tool provided in Debian's `whois`
+  package; the password will not be set or changed if this parameter is not
+  set);
+- `force_password` (boolean) (default: `False`): whether or not the user's
+  password should be set to `password`'s value even if the user was not newly
+  created;
+- `pubkey` (string) (mandatory): one or several SSH public keys, separated
+  by new lines; SSH key not listed will be removed from user's `authorized_keys`
+  file.
+
+Example:
+
+```
+deflect_ssh_users:
+  - login: bob
+    sudo: True
+    password: $6$TYZFllHTbDCBn$Lph3t5eaaM3YTcqKeowf38z5BOxYVgK.a9nrkpaozmDXymX934o/NpGY1WI1wjfbgrPK0keFxPRun93FbdSi81
+    pubkey: ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDEYa5C/1gMUTM9x4MnhhPW1UttT8VFL3/eMxDD6UuPy0MoREdjShdd6v+sA3ivOx8OAu5wt9mgBXcnl6LYaLw9ivPQ6vN3wrFD3eHVsJnRyl6YW4nn9ADeNH+jI96OF7nyUg2D2tEdG+v2BW74hhUePEefZsJV7MfhHB49bynJrpegV0lPeamczddhtMgorgbSyLodqs8TO4o+xvjJPFPD3CriCq38tmBZ02XWq/H8ac/13pwNsQdHTjHGt5fO7YJXoz5KBRUXDvHkf435AMBGc1jCx8PGn4jMcY1vfnlhWoU8/E84kxgykcsTAK16fuaCqJVlFJ/rhiinYqphtBfT
```

Comment:
This is not how i would distribute public keys or "shared secrets" in general. I understand this is the "public" key and all that but it's really something for https://www.consul.io/ or https://docs.ansible.com/ansible/2.4/vault.html

again as in interim checkin i'm vaguely okay w/ this. 

1. why are you still using `ssh-rsa` keys? is it 1999? ecdsa https://linux-audit.com/using-ed25519-openssh-keys-instead-of-dsa-rsa-ecdsa/ 
2. using ansible vault will be better in the long run to not "hardcode" the pub-keys here but use ansible vault to replace the keys here programmatically from a trusted managed source.

--timball